### PR TITLE
Refactor auth

### PR DESCRIPTION
**Pull request by iammarkps**

### CODE REVIEW

File: src/components/UserContext.tsx [hunk 1]
```diff
@@ -1,4 +1,8 @@
-import React, { useContext } from 'react'
+import React, { useContext, useReducer, useEffect } from 'react'
+import useSWR, { mutate } from 'swr'
+import { fetchFromFirebase } from 'utils/fetcher'
+import firebase from 'firebase'
```
Comment (win11905) [inline src/components/UserContext.tsx:4]: Why do your import firebase from 'firebase'? 
Could it be 'lib/firebase'?

Comment (iammarkps) [reply src/components/UserContext.tsx]: Fixed

File: src/components/UserContext.tsx [hunk 2]
```diff
@@ -50,3 +62,53 @@ export const reducer = (state: UserState, action: UserAction): UserState => {
 export const UserStateContext = React.createContext<UserState>(initialState)
 
 export const useUser = () => useContext(UserStateContext)
+
+export default function UserContextComp({ children }) {
```
Comment (win11905) [inline src/components/UserContext.tsx:66]: Why don't you use ES6 functions?
it can be replaced by `export default ({ children }) => {`

**Comment by win11905** [comment]
In `UserContext.tsx`, some functions don't need to export because it necessarily uses only in this file.

File: src/components/UserContext.tsx (hunk 2 above)
Comment (iammarkps) [reply src/components/UserContext.tsx:66]: This way in stack trace will show up UserContextComp not default without name. You can see from Dan Abramov's twitter

**Pull request merged**